libs/game/physics.ts

```diff
@@ -379,62 +518,12 @@ class ArcadePhysicsEngine extends PhysicsEngine {
 
         if (xDiff !== Fx.zeroFx8) {
             const right = xDiff > Fx.zeroFx8;
-            const x0 = Fx.toIntShifted(
-                Fx.add(
-                    right ?
-                        Fx.add(hbox.right, Fx.oneFx8)
-                        :
-                        Fx.sub(hbox.left, Fx.oneFx8),
-                    Fx.oneHalfFx8
-                ),
-                tileScale
-            );
-            const collidedTiles: sprites.StaticObstacle[] = [];
+            const collisionHandlers = this.checkHorizontalCollisions(right, xDiff, yDiff, movingSprite, tm);
 
```

Comment:
I actually had these pulled out into helpers at one point, but inlining them had about a 2-3 fps difference when playing jumpy platformer on meowbit (which was about 20 fps or so if I'm remembering right)

Do you have a minute to compare fps with this change vs current master? I can take a look in a bit, went out to grab something to eat. For what it's worth, there have been a few improvements in this area since I wrote this part so it might not make a difference anymore

Reply:
Yes I can do it! is there a good way to do it though 😅

(i'm glad you're eating)

Reply:
In the standard menu there's a "show fps" button that turns on some stats at the bottom of the screen - not perfect but good for a rough estimate like what's needed here ~

Reply:
In the standard menu there's a "show fps" button that turns on some stats at the bottom of the screen - not perfect but good for a rough estimate like what's needed here ~

Reply:
If you put this line in a game:

```ts
game.stats = true;
```

It will print out the FPS to the screen on hardware.

I know we have some physics tests checked in, but I don't know if they are perf related: https://github.com/microsoft/pxt-arcade/tree/master/docs/physics-tests

Reply:
Yeah those ones are mainly behavioral, the shapes one is a light perf test but not consistant at the moment

Reply:
oo cool so with my test game (https://makecode.com/_WzME6j8uUewe)

Running 50 sprites on **beta**:
(A) Going left: 10 - 15 fps
(B) Bounce: 4.5 - 5.7 fps

Running 50 sprites with **these changes**:
(A) Going left: 10.4 - 12.9 fps
(B) Bounce: 3.3 - 4.9 fps

It looks like it makes it slower :\  or at least it's harder to see the faster numbers with my changes